src/core/Client.ts

```diff
@@ -82,32 +69,37 @@ export class Client<TABITable extends ABITable> {
    *     { account },
    * );
    */
-  public async submitTransaction(
+  public async submitTransaction(args: {
+    signer: Account,
     payload: EntryPayload,
-    options: EntryOptions,
-  ): Promise<TransactionResponse> {
-    const rawTxn = await this.generateRawTxn(payload, options);
-
-    // Sign the raw transaction with account's private key
-    const bcsTxn = AptosClient.generateBCSTransaction(options.account, rawTxn);
+  }): Promise<CommittedTransactionResponse> {
+    const transaction = await this.client.build.transaction({
+      sender: args.signer.accountAddress.toString(),
+      data: args.payload,
+    });
 
     // Submit the transaction
-    const transactionRes = await this.client.submitSignedBCSTransaction(bcsTxn);
+    const transactionRes = await this.client.signAndSubmitTransaction({
+      signer: args.signer,
+      transaction,
+    });
 
     // Wait for the transaction to finish
     // throws an error if the tx fails or not confirmed after timeout
-    await this.client.waitForTransaction(transactionRes.hash, {
-      timeoutSecs: 120,
-      checkSuccess: true,
+    return await this.client.waitForTransaction({
+      transactionHash: transactionRes.hash,
+      options: {
```

Comment:
good to have the default, but shall we also allow overriding the options when necessary?

Reply:
this applies to other function calls - lmk wdyt